src/components/measurements/index.js

```diff
@@ -0,0 +1,279 @@
+        orderedFields.forEach((field) => {
+          newData.set(fields.get(field).title, data[field]);
+        });
+      } else if (dataType === "mean") {
+        // Handle mean and standard deviation data
+        newData.set("mean", data.mean.toFixed(2));
+        newData.set("standard deviation", data.standardDeviation ? data.standardDeviation.toFixed(2) : "N/A");
+      } else {
+        // Catch unknown data types
+        console.error(`"Unknown data type for hover panel: ${dataType}`);
+        // Display provided data without extra ordering or parsing
+        Object.entries(data).forEach(([key, value]) => newData.set(key, value));
+      }
+      newHoverData = {
+        mouseX,
+        mouseY,
+        containerId: svgContainerDOMId,
+        data: newData
+      };
+    }
+    setHoverData(newHoverData);
+  }, [fields]);
+
+  useEffect(() => {
+    setPanelTitle(`${title || "Measurements"} (grouped by ${fields.get(groupBy).title})`);
+  }, [setPanelTitle, title, fields, groupBy]);
+
+  // Draw SVG from scratch
+  useEffect(() => {
+    clearMeasurementsSVG(d3Ref.current);
+    drawMeasurementsSVG(d3Ref.current, svgData, handleHover);
+  }, [svgData, handleHover]);
+
+  // Color the SVG & redraw color-by means when SVG is re-drawn or when colors have changed
+  useEffect(() => {
+    colorMeasurementsSVG(d3Ref.current, treeStrainColors);
+    drawMeansForColorBy(d3Ref.current, svgData, treeStrainColors, handleHover);
+  }, [svgData, treeStrainColors, handleHover]);
+
+  // Display raw/mean measurements when SVG is re-drawn, colors have changed, or display has changed
+  useEffect(() => {
+    changeMeasurementsDisplay(d3Ref.current, display);
+  }, [svgData, treeStrainColors, display]);
+
+  useEffect(() => {
+    toggleDisplay(d3Ref.current, "overallMean", showOverallMean);
+  }, [svgData, showOverallMean]);
+
+  useEffect(() => {
+    toggleDisplay(d3Ref.current, "threshold", showThreshold);
+  }, [svgData, showThreshold]);
+
+  const getSVGContainerStyle = () => {
+    return {
+      overflowY: "auto",
+      position: "relative",
+      height: height,
+      width: width
+    };
+  };
+
+  return (
+    <>
+      {showLegend &&
+        <ErrorBoundary>
+          <Legend right width={width}/>
+        </ErrorBoundary>
+      }
+      <div id={svgContainerDOMId} style={getSVGContainerStyle()}>
+        {hoverData &&
+          <HoverPanel
+            hoverData={hoverData}
+          />
+        }
+        <svg
+          id="d3MeasurementsSVG"
+          width="100%"
+          ref={d3Ref}
+        />
+      </div>
+    </>
+  );
+};
+
+const Measurements = ({height, width, showLegend}) => {
+  const measurementsLoaded = useSelector((state) => state.measurements.loaded);
+  const measurementsError = useSelector((state) => state.measurements.error);
+
+  const [title, setTitle] = useState("Measurements");
+
+  return (
+    <Card title={title}>
```

Comment:
I think the title CSS (in `<Card>`) needs to be changed as when in grid mode the panels can be displayed on top of each other due to long title names

![image](https://user-images.githubusercontent.com/8350992/155632449-06f634d7-6b5d-475f-86a0-07152c72c1c1.png)

Reply:
Added as todo in #1463